src/utils/thumbnailCreator.ts

```diff
@@ -0,0 +1,51 @@
+export interface Thumbnail {
+  from: number;
+  to: number;
+  imgUrl: string;
+}
+export const SCALE_FACTOR = 1;
+export default async function* extractThumbnails(
+  videoUrl: string,
+  numThumbnails: number
+): AsyncGenerator<Thumbnail, Thumbnail> {
+  const canvas = document.createElement("canvas");
+  const ctx = canvas.getContext("2d");
+  if (!ctx) return { from: -1, to: -1, imgUrl: "" };
+  const video = document.createElement("video");
+  video.src = videoUrl;
+  video.crossOrigin = "anonymous";
+
+  // Wait for the video metadata to load
+  const metadata = await new Promise((resolve, reject) => {
```

Comment:
```suggestion
  await new Promise((resolve, reject) => {
```

metadata is not used, linter is complaining.